src/app/user/(logged-in)/my-account/page.tsx

```diff
@@ -1,27 +1,101 @@
 "use server";
 
-import { BarkH1 } from "@/components/bark/bark-typography";
 import { getAuthenticatedUserActor } from "@/lib/auth";
 import { RoutePath } from "@/lib/route-path";
-import { UserPii } from "@/lib/data/db-models";
 import { redirect } from "next/navigation";
+import { Button, buttonVariants } from "@/components/ui/button";
+import { IMG_PATH } from "@/lib/image-path";
+import { getMyLatestCall } from "@/lib/user/actions/get-my-latest-call";
+import { getMyAccount } from "@/lib/user/actions/get-my-account";
+import { format, formatDistanceStrict } from "date-fns";
+import { BarkH1, BarkH4 } from "@/components/bark/bark-typography";
+import Link from "next/link";
+import Image from "next/image";
+
+import { capitalize } from "lodash";
 
 export default async function Page() {
   const actor = await getAuthenticatedUserActor();
   if (!actor) {
     redirect(RoutePath.USER_LOGIN_PAGE);
   }
-  const ownPii: UserPii | null = await actor.getOwnUserPii();
-  if (!ownPii) {
+  const account = await getMyAccount(actor);
+  if (!account) {
     redirect(RoutePath.USER_LOGIN_PAGE);
   }
+  const {
+    userId,
+    userCreationTime,
+    userResidency,
+    ownPii: { userName, userEmail, userPhoneNumber },
+  } = account;
+
+  let latestCall = (await getMyLatestCall(actor))?.callCreationTime;
+  const latestCallText = latestCall
+    ? formatDistanceStrict(latestCall, new Date(), { addSuffix: true })
+    : "N.A";
+
+  const userCreationTimeText = format(userCreationTime, "dd MMM yyyy");
```

Comment:
What timezone would be used to format the creation time text?

Reply:
Saw the `bark-time` formatting accounted for timezone! So using `SINGAPORE_TIME_ZONE`

54b6cb146d9c7203f510d1e8bed741990e011297